lib/tasks/retention_instructions.rake

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# $ bundle exec rake "retention_instructions:backfill[batch_size]"
+
+require_relative '../../app/helpers/retention_instruction_helper'
+
+namespace :retention_instructions do
+
+  include RetentionInstructionHelper
+  desc 'Backfill retention instructions'
+
+  task :backfill, [:batch_size] => :environment do |_, args|
+    args.with_defaults(batch_size: 1000)
+    batch_size = args[:batch_size].to_i
+
+    puts "Backfilling retention instructions with batch size: #{batch_size}..."
+
+    ActiveRecord::Base.transaction do
+      saved_count = 0
+      Labware.where(retention_instruction: nil).find_each(batch_size: batch_size) do |labware|
+        saved_count = process_labware(labware, saved_count)
+      end
+      puts "Backfilled retention instructions for #{saved_count} labware items."
+    end
+
+  end
+
+  # rubocop:todo Metrics/AbcSize
+  # rubocop:todo Metrics/MethodLength
+  def process_labware(labware, saved_count)
+    return saved_count unless labware.custom_metadatum_collection.present? &&
+      labware.custom_metadatum_collection.metadata['retention_instruction'].present?
+
+    labware.retention_instruction = find_retention_instruction_key_for_value(
+      labware.custom_metadatum_collection.metadata['retention_instruction']
+    ).to_sym
+
+    begin
+      labware.custom_metadatum_collection.custom_metadata.where(key: 'retention_instruction').find_each(&:destroy!)
+      saved_count += 1 if labware.save!
+      saved_count
```

Comment:
On failure, labware.save! will raise an exception. On success, it will return true. 

On failure, the last value generated in the block  by the augmented assignment will be nil, but the code jumps to the rescue block. On success, the last value generated in the block will be the result of the augmented assignment.

saved_count statement in the separate line looks redundant.

Reply:
So the idea is to return the updated count after the script is done without errors. So I think it needs to be returned from this function, right? Because only incrementing would not suffice as the saved_count initialized within the transaction is copied into this function's stack. We can return it at the end of the function though.